ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/query/json/ContractStateVaultJsonFactory.java

```diff
@@ -0,0 +1,80 @@
+package net.corda.v5.ledger.utxo.query.json;
+
+import net.corda.v5.application.marshalling.JsonMarshallingService;
+import net.corda.v5.ledger.utxo.ContractState;
+import net.corda.v5.ledger.utxo.UtxoLedgerService;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Implement a {@link ContractStateVaultJsonFactory} to create a JSON representation of a state to store alongside
+ * the state in the vault. This JSON representation can be used to query states from flows using
+ * {@link UtxoLedgerService#query}.
+ * <p>
+ * Classes that implement {@link ContractStateVaultJsonFactory} will be executed in a hierarchical way. That means
+ * every factory that belongs to a given state or that given state's ancestors will be executed and the end result
+ * will be a combined JSON. The combined JSON will be keyed by the state type specified by the factory.
+ * <p>
+ * Please note that only one factory can be registered for a state type.
```

Comment:
After the part where we talk about the keys, can we also state, that if a "query by state" type is needed for a particular state type, then there MUST be a factory registered, even if it returns an empty string/json object.

Reply:
Try word it nicer than I did